CSC8503/InventoryBuffSystem/PickupGameObject.cpp

```diff
@@ -0,0 +1,130 @@
+#include "PickupGameObject.h"
+#include "StateTransition.h"
+#include "StateMachine.h"
+#include "State.h"
+#include "PhysicsObject.h"
+#include "Vector3.h"
+#include "random"
+#include "InventoryBuffSystem.h"
+
+using namespace NCL;
+using namespace CSC8503;
+using namespace InventoryBuffSystem;
+
+PickupGameObject::PickupGameObject(std::map<GameObject*, int>* mPlayerObjectToPlayerNoMap, float initCooldown) {
+	mCooldown = 0.0f;
+	mInitCooldown = initCooldown;
+	mPlayerObjectToPlayerNoMap = mPlayerObjectToPlayerNoMap;
+
+	mStateMachine = new StateMachine();
+	State* WaitingState = new State([&](float dt) -> void
+		{
+			this->Waiting(dt);
+		}
+	);
+	State* GoUnderState = new State([&](float dt) -> void
+		{
+			this->GoUnder(dt);
+		}
+	);
+	State* GoOverState = new State([&](float dt) -> void
+		{
+			this->GoOver(dt);
+		}
+	);
+
+	mStateMachine->AddState(WaitingState);
+	mStateMachine->AddState(GoUnderState);
+	mStateMachine->AddState(GoOverState);
+
```

Comment:
Is there any way to do this in multiple functions as opposed to a really long constructor?